### ISSUE TITLE
Enable Taxonomy Tags with local_only=0

### ISSUE THREAD

**Issue by github-germ** [MISP/MISP#7999]
2.4.151
Related:
* https://github.com/MISP/MISP/issues/7876
* https://github.com/MISP/MISP/pull/7892

We upgraded to 2.4.151 and have an issue with our PyMISP apps failing with the tagging process as follows.

We've added a custom taxonomy and enabled all contained tags resulting in `Tags.local_only=1` (in the db) for all these new tags. This causes our existing PyMISP app calls to `pymisp.tag()` to fail with `{"0":"Invalid Tag. This tag can only be set as a local tag."}'}`. 

**Question**: Do we need to revise all PyMISP app calls to `tag` to set `local=True` or is there a way to make these custom taxonomy tags all global when installing/enabling?

**Comment by righel** [comment]
Hello @github-germ 
I experienced this issue in a VM installation last week, after your report I've been trying to reproduce it without success.
Is this reproducible for you in an up-to-date MISP instance (2.4.151 / b869ff91549436ce189be9ff76aa4d1581384f3f) ?

If so, could you share the exact steps to reproduce?

Thanks.

**Comment by github-germ** [comment]
Hi @righel -- Thanks...

Yes, we are on 2.4.151. Our scenario in adding a custom taxonomy uses a shell script that does the following:
1. Add taxonomy dir with `machinetag.json` to `/var/www/MISP/app/files/taxonomies` and into Python `.../site-packages/pytaxonomies/data/misp-taxonomies`
2. Revise `MANIFEST.json` adding the taxonomy into both MISP and `pytaxonomies`
3. Invoke our own PyMISP script that does:
  a. `pymisp.update_taxonomies()`
  b. get the new `taxonomy_id` based on `namespace` with `pymisp.taxonomies()`
  c. `pymisp.enable_taxonomy(taxonomy_id)`
  d. `misp.enable_taxonomy_tags(taxonomy_id)`

And now all the tags within that taxonomy have `local_only=1`

So, to workaround this, I've added another step in the PyMISP script:
* 3e. loop through all tags from `pymisp.get_taxonomy(taxonomy_id)` setting `local_only=False` via `pymisp.update_tag`

**Comment by righel** [comment]
Hi @github-germ 
I did the following:
1. Add `test-local-only` directory to:
    * `/var/www/MISP/app/files/taxonomies` 
    * `/var/www/MISP/venv/lib/python3.8/site-packages/pytaxonomies/data/misp-taxonomies`
2. Add a `machinetag.json` file inside both new directories with the following content:
```json
    {
      "values": [
        {
          "entry": [
            {
              "expanded": "Test Entry for local_only behavior.",
              "value": "test-local-only",
              "numerical_value": 1
            }
          ],
          "predicate": "test-local-only-predicate"
        }
      ],
      "predicates": [
        {
          "description": "Test Predicate for local_only behavior.",
          "expanded": "Test predicate for local_only behavior.",
          "value": "test-local-only-predicate"
        }
      ],
      "version": 1,
      "refs": [],
      "description": "Test Taxonomy for local_only behavoir.",
      "namespace": "test-local-only"
    }
```

3. Added the following entry to `/var/www/MISP/venv/lib/python3.8/site-packages/pytaxonomies/data/misp-taxonomies/MANIFEST.json`:
```json
   {
      "description": "Test Taxonomy for local_only behavior.",
      "name": "test-local-only",
      "version": 1
    }
```
4. Run the following Python test script:
    ```python
    #!/var/www/MISP/venv python3.8
    # -*- coding: utf-8 -*-
    
    from pymisp import PyMISP
    from pprint import pprint
    
    pymisp = PyMISP(
        "https://localhost/",
        "YOUR_API_KEY",
        False,
        'json'
    )
    
    pymisp.update_taxonomies()
    
    taxonomies = pymisp.taxonomies()
    
    for taxonomy in taxonomies:
        if taxonomy["Taxonomy"]["namespace"] == "test-local-only":
            pymisp.enable_taxonomy(taxonomy["Taxonomy"]["id"])
            pymisp.enable_taxonomy_tags(taxonomy["Taxonomy"]["id"])
    pprint(pymisp.direct_call("https://localhost/tags/search", {"tag":  "%local-only%"}))
    ```
    **Output:**
    ```json
    [
       {
          "Tag":{
             "colour":"#ffa800",
             "exportable":true,
             "hide_tag":false,
             "id":"1138",
             "is_custom_galaxy":false,
             "is_galaxy":false,
             "local_only":false,
             "name":"test-local-only:test-local-only-predicate=\"test-local-only\"",
             "numerical_value":"1",
             "org_id":"0",
             "user_id":"0"
          },
          "Taxonomy":{
             "description":"Test Taxonomy for local_only behavior.",
             "enabled":true,
             "exclusive":false,
             "id":"146",
             "namespace":"test-local-only",
             "required":false,
             "version":"1"
          },
          "TaxonomyPredicate":{
             "TaxonomyEntry":[
                {
                   "colour":"None",
                   "description":"None",
                   "expanded":"Test Entry for ""local_only behavior.",
                   "id":"7351",
                   "numerical_value":"1",
                   "taxonomy_predicate_id":"974",
                   "value":"test-local-only"
                }
             ],
             "colour":"None",
             "description":"Test Predicate for local_only ""behavior.",
             "exclusive":false,
             "expanded":"Test predicate for local_only behavior.",
             "id":"974",
             "numerical_value":"None",
             "taxonomy_id":"146",
             "value":"test-local-only-predicate"
          }
       }
    ]
    ```

This results in the tag imported with the `local_only` is set to `false`. Could not reproduce the issue.

Let me know if I'm missing something in any of the steps. 
As I did experience this issue (last week) I would suggest you run the updates again, version 2.4.151 was re-tagged at some point and you may have a previous commit that has some kind of bug related to this.

**Comment by github-germ** [comment]
Hey @righel -- Your test case matches my approach. Indeed, we are still on https://github.com/MISP/MISP/commit/3abe0594047500a6c11a9a61e26406e3bc508d5e -- our corporate policies only allow us to deploy "official" releases. I would agree that some pull after the official 2.4.151 looks to have resolved this issue. I'd be curious to see what fixed it. However, for now I've added the code to flip the `local_only` setting -- so we are OK.

**Comment by cybgit** [comment]
We appear to have this issue after upgrading from 2.4.143 to 2.4.153. An existing (custom) taxonomy now has all tags set to local only.  Any new tags we add to the taxonomy appear to be set as global which is good but all our previous tags in the taxonomy have this local only flag set. 

If i trace the UI calls when updating a tag manually to unset local only i can see in the request that it sets `<html>data[Tag][local_only] | "0"` yet i can't see an API parameter for the edit tag endpoint that has an option to set this local_only flag. I'm looking here. https://<misp_server>/servers/openapi#operation/editTag

**Comment by adulau** [comment]
This should be fixed in https://github.com/MISP/MISP/commit/e73950d0d24e2ab9d1557bc0ba601072fa54d9b9 - the fix is in 2.4 branch. Feedback is welcome. I'll close the issue. If the issue persists, feel free to reopen.

**Comment by github-germ** [comment]
Thanks!
